src/core/cgroup.c

```diff
@@ -397,17 +397,20 @@ static void cgroup_xattr_apply(Unit *u) {
 
 static int lookup_block_device(const char *p, dev_t *ret) {
         struct stat st = {};
+        dev_t st_rdev;
```

Comment:
pleas just call this `rdev`, and plesae move the "struct stat st" definition into the inner `if` block that actually calls stat(), then copy out the relevant fields st_stdev and st.mode from there in that inner block. (and yes, this also means definining a new variable `mode_t mode` here...

i.e. the "struct stat" thing should be internal to that inner block now, and we copy out everything before using it, as apparently the structure is not suitable for storing our own data in otherwise.